IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetSsoCookiesResponse.m

```diff
@@ -0,0 +1,163 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#import "MSIDBrokerOperationGetSsoCookiesResponse.h"
+#import "MSIDPrtHeader.h"
+#import "MSIDDeviceHeader.h"
+#import "MSIDJsonSerializableFactory.h"
+#import "MSIDJsonSerializableTypes.h"
+#import "MSIDJsonSerializer.h"
+#import "NSString+MSIDExtensions.h"
+#import "NSDictionary+MSIDExtensions.h"
+
+static NSString *const MSID_PRT_HEADERS = @"prt_headers";
+static NSString *const MSID_DEVICE_HEADERS = @"device_headers";
+
+@implementation MSIDBrokerOperationGetSsoCookiesResponse
+
++ (void)load
+{
+    [MSIDJsonSerializableFactory registerClass:self forClassType:self.responseType];
+}
+
++ (NSString *)responseType
+{
+    return MSID_JSON_TYPE_BROKER_OPERATION_GET_SSO_COOKIES_RESPONSE;
+}
+
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+    
+    if (self)
+    {
+        if (![json msidAssertType:NSString.class ofKey:@"sso_cookies" required:NO error:error])
+        {
+            return nil;
+        }
+        
+        NSString *ssoCookiesString = json[@"sso_cookies"];
+        if ([NSString msidIsStringNilOrBlank:ssoCookiesString])
+        {
+            return self;
+        }
+        
+        NSDictionary *ssoCookiesJson = [ssoCookiesString msidJson];
+        NSError *convertPrtHeaderError = nil;
+        NSError *convertDeviceHeaderError = nil;
+        _prtHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_PRT_HEADERS error:&convertPrtHeaderError];
+        _deviceHeaders = [self convertToCredentialHeaderObjectFrom:ssoCookiesJson credentialName:MSID_DEVICE_HEADERS error:&convertDeviceHeaderError];
+        
+        if(convertPrtHeaderError && convertDeviceHeaderError)
+        {
+            if(error)
+            {
+                *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, [NSString stringWithFormat:@"Cannot convert credential headers from json response. PrtHeader error is %@. Device header error is %@", convertPrtHeaderError.description, convertDeviceHeaderError.description], nil, nil, nil, nil, nil, YES);
+            }
+            
+            self.success = NO;
+        }
+        else if (convertPrtHeaderError)
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Cannot convert Prt Header from json response. Error: %@", convertPrtHeaderError.description);
+        }
+        else
+        {
+            MSID_LOG_WITH_CTX(MSIDLogLevelInfo, nil, @"Cannot convert device Header from json response. Error: %@", convertDeviceHeaderError.description);
```

Comment:
Seems we always fall into this if statement. Should it be `if (convertDeviceHeaderError)`?

Reply:
good catch